## **EXPLORATORY DATA ANALYSIS**
### Título

### ***ÍNDICE***
1. Importación de librerías
2. Carga de datos
3. Selección de columnas (Creamos un dataframe nuevo)
4. Tipos de datos y valores nulos
5. Decodificación de variables
6. Renombrar columnas


>  **1.**

In [274]:
# Importar librerías

import numpy as np
import pandas as pd

import matplotlib.pyplot as plt


>  **2.**

In [275]:
# Carga de datos DataFrame principal

df = pd.read_csv("data/EILU_2019.csv",sep="\t")

C:\Users\beatr\AppData\Local\Temp\ipykernel_25576\4148141566.py:3: DtypeWarning: Columns (119) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("data/EILU_2019.csv",sep="\t")


In [276]:
# Obtenemos un vistazo del df, sus columnas y el formato de sus filas
df

IDENT  SEXO  EDAD  NACIO NACIO1 NACIO2  PAIS_NACI    TITU  RAMA  \
0          1     1     1      1                        1   11101     3   
1          2     1     1      1                        1   11101     3   
2          3     1     1      1                        1   11101     3   
3          4     1     1      1                        1   11101     3   
4          5     1     1      1                        1   11101     3   
...      ...   ...   ...    ...    ...    ...        ...     ...   ...   
31646  31647     2     2      1                        1  109999     3   
31647  31648     2     2      1                        1  109999     3   
31648  31649     1     3      1                        1  109999     3   
31649  31650     1     1      1                        1  109999     3   
31650  31651     1     3      1                        1  109999     3   

       AMBITO  ...  ANHCUR4  DENCUR5  DURCUR5  RESCUR5  ANHCUR5  DENCUR6  \
0          11  ...                                                         
1          11  ...                                                         
2          11  ...                                                         
3          11  ...                                                         
4          11  ...                                                         
...       ...  ...      ...      ...      ...      ...      ...      ...   
31646     109  ...                                                         
31647     109  ...                                                         
31648     109  ...                                                         
31649     109  ...                                                         
31650     109  ...                                                         

       DURCUR6  RESCUR6  ANHCUR6     FACTOR  
0                                  9.894742  
1                                  9.894742  
2                                 10.965467  
3                                 10.859689  
4                                 10.859689  
...        ...      ...      ...        ...  
31646                              1.780764  
31647                              1.780764  
31648                              3.038041  
31649                              3.038041  
31650                              2.311008  

[31651 rows x 299 columns]

In [277]:
df.shape

(31651, 299)

Observamos que tiene **31651 filas**. Es un nivel alto de datos y no todos los registros nos interesarán para el estudio. Más adelante filtraremos aquellas filas que no nos interesan. En cuanto a las columnas, un valor de **299 columnas** es grande y tenemos que tener en cuenta aquellas que sí nos serán de utilidad para comprobar nuestra hipótesis y formular las conclusiones, es decir, aquellas que realmente tienen valor para nuestro análisis exploratorio

>  **3.**

Para saber qué columnas aportan valor a nuestro estudio recurrimos a un segundo archivo: "data/dr_EILU_2019.xlsx" Nuestro conjunto de datos completo se compone de dos archivos: un csv con los registros codificados de 31561 individuos y el archivo excel (del que más adelante hablaremos) con la codificación y diseño de los datos. Este archivo nos puede servir de guía para comprender los campos y valores del DataFrame principal.

In [278]:
# Cargamos el diseño del conjunto de datos que contiene la información de las columnas codificadas

cod = pd.read_excel("data/dr_EILU_2019.xlsx",header=1,usecols = 'A:I') # O eliminar las col unnamed
cod

Variable Diccionario de la variable  Longitud Tipo  Decimales  Posición  \
0      IDENT                        NaN         5    N        NaN       1.0   
1       SEXO                      TSEXO         1    A        NaN       6.0   
2       EDAD                      TEDAD         1    A        NaN       7.0   
3      NACIO                     TNACIO         1    A        NaN       8.0   
4     NACIO1                    TNACIOP         1    A        NaN       9.0   
..       ...                        ...       ...  ...        ...       ...   
295  DURCUR6                    TDrCurs         1    A        NaN     371.0   
296  RESCUR6                    TReCurs         1    A        NaN     372.0   
297  ANHCUR6                    TAnCurs         4    A        NaN     373.0   
298   FACTOR                        NaN        11    N        6.0     377.0   
299    TOTAL                        NaN       387  NaN        NaN       NaN   

     Orden Diccionario ubicado en la hoja…  \
0      1.0                             NaN   
1      2.0                         Tablas1   
2      3.0                         Tablas1   
3      4.0                         Tablas1   
4      5.0                         Tablas1   
..     ...                             ...   
295  296.0                         Tablas2   
296  297.0                         Tablas2   
297  298.0                         Tablas2   
298  299.0                             NaN   
299    NaN                             NaN   

                                           Descripción  
0                   Nº de identificación de la persona  
1                                                 Sexo  
2                        Grupo de edad (a 31-dic-2019)  
3                                         Nacionalidad  
4                              País de nacionalidad 1   
..                                                 ...  
295  Duración del sexto curso de formación del SEPE...  
296  Resultado del sexto curso de formación del SEP...  
297  Año de finalización del sexto curso de formaci...  
298              Factor de elevación (seis decimales )  
299                                                NaN  

[300 rows x 9 columns]

Este archivo, se trata de un formato excel con varías páginas. En la primera hoja, contiene un listado con toda las variables y su descripción. Además, contiene una variable que te indica en qué página del excel se encuentra la codificación de cada variable.

Tras estudiar los diferentes campos, nos quedamos con las siguientes columnas:

In [279]:
# Limpieza de datos
students = df[['IDENT','SEXO','EDAD','NACIO','TITU','RAMA','AMBITO','T_UNIV','DISCA','ESTUDIOS_PADRE','ESTUDIOS_MADRE',
                  'EST_B8','SAT1','SAT2','EST_B11_1','EST_B13_AMB1','EST_B11_2','EST_B11_3','EST_B11_4','EST_B11_5','EST_B11_6','EST_B12',
                  'EST_B16','EST_B24','EST_B28','IDIOMAS', 'IDI_MT1', 'IDI_MT2','TIC','IDIOMA1','NIV_ID1','TIPOACREIDI1',
                  'IDIOMA2','NIV_ID2','TIPOACREIDI2','IDIOMA3','NIV_ID3','TIPOACREIDI3','TRBPRN1','TRBPRN2','TRINAC_D4',
                  'TRINAC_D5','TRINAC_D6','TRPARA_D5','TRPARA_D6','TRAC_D9','LUG_TRAB_PAIS','SIT_PRO','JORNADA','TR_TAM',
                  'TR_SUELDO','TR_D19','TR_D21', 'HL_E1','HL_E2','HL_E3','HL_E5','HATR_E11','HATR_E12','P_CONTI','PR_SIT_PR',
                  'PR_JOR','PR_NIVEL','PR_AREA','PR_CONOC','PR_SUELDO','HATR_E27','NOTRAB_BUS','NOTRAB_BUSM','NOTRAB_BUST']]

In [280]:
students

IDENT  SEXO  EDAD  NACIO    TITU  RAMA  AMBITO  T_UNIV  DISCA  \
0          1     1     1      1   11101     3      11       2      2   
1          2     1     1      1   11101     3      11       2      2   
2          3     1     1      1   11101     3      11       1      2   
3          4     1     1      1   11101     3      11       1      2   
4          5     1     1      1   11101     3      11       1      2   
...      ...   ...   ...    ...     ...   ...     ...     ...    ...   
31646  31647     2     2      1  109999     3     109       1      2   
31647  31648     2     2      1  109999     3     109       1      2   
31648  31649     1     3      1  109999     3     109       1      2   
31649  31650     1     1      1  109999     3     109       1      2   
31650  31651     1     3      1  109999     3     109       1      2   

       ESTUDIOS_PADRE  ...  PR_SIT_PR  PR_JOR  PR_NIVEL PR_AREA  PR_CONOC  \
0                   2  ...          1       1         3       2         1   
1                   3  ...          3       1         3       2         1   
2                   3  ...          3       2         4       3         2   
3                   7  ...          3       2         4       3         2   
4                   8  ...          3       2         3       2         1   
...               ...  ...        ...     ...       ...     ...       ...   
31646               9  ...          2       2         2       3         1   
31647               6  ...          2       2         3       1         2   
31648               3  ...          2       2         3       1         1   
31649               8  ...          2       1         3       1         9   
31650               4  ...          2       2         3       2         1   

      PR_SUELDO  HATR_E27  NOTRAB_BUS  NOTRAB_BUSM  NOTRAB_BUST  
0             3         1                                        
1             3         1                                        
2             3         1                                        
3             2         1                                        
4             3         2                                        
...         ...       ...         ...          ...          ...  
31646         3         1                                        
31647         3         2                                        
31648         3         1                                        
31649         9         1                                        
31650         3         1                                        

[31651 rows x 70 columns]

La información de las columnas seleccionadas es la siguiente:

In [281]:
# Fijamos la columna 'Variable' como índice de este DataFrame, con el fin de poder buscar más fácilmente por nombres de variables.
cod.set_index('Variable',inplace=True)

In [282]:
cod.index

Index(['IDENT', 'SEXO', 'EDAD', 'NACIO', 'NACIO1', 'NACIO2', 'PAIS_NACI',
       'TITU', 'RAMA', 'AMBITO',
       ...
       'DENCUR5', 'DURCUR5', 'RESCUR5', 'ANHCUR5', 'DENCUR6', 'DURCUR6',
       'RESCUR6', 'ANHCUR6', 'FACTOR', 'TOTAL'],
      dtype='object', name='Variable', length=300)

In [283]:
columns_list = students.columns.to_list() # Lista con el nombre de las columnas de nuestro DataFrame

# Por cada columna del DataFrame students nos imprime la descripción de la misma.

descripciones = [] # Lista de listas donde cada valor será el par [varialbe,descripción]

for variable in cod.index:
    if variable in columns_list:
        descripciones.append([variable, cod.loc[variable]['Descripción']])

In [284]:
for variable, descripcion in descripciones:
    print(f"{variable} : {descripcion}")

IDENT : Nº de identificación de la persona
SEXO : Sexo
EDAD : Grupo de edad (a 31-dic-2019)
NACIO : Nacionalidad
TITU : Titulación
RAMA : Rama de conocimiento de la titulación
AMBITO : Ámbito de estudio
T_UNIV : Tipo de Universidad
DISCA : Tiene discapacidad reconocida superior al 33%
ESTUDIOS_PADRE : Nivel de formación máximo alcanzado por el padre
ESTUDIOS_MADRE : Nivel de formación máximo alcanzado por la madre
EST_B8 : Motivo principal por el que ha realizado sus estudios
SAT1 : Si tuviese que empezar volvería a cursar estudios universitarios
SAT2 : Si tuviese que empezar volvería a estudiar la misma titulación
EST_B11_1 : Otros estudios: Grado / Diplomatura / Licenciatura o equivalente
EST_B11_2 : Otros estudios: Máster universitario
EST_B11_3 : Otros estudios: Doctorado universitario
EST_B11_4 : Otros estudios: Estudios artísticos superiores (artes plásticas, diseño, música y danza, arte dramático o conservación y restauración de bienes culturales)
EST_B11_5 : Otros estudios: Cic

Con el fin de tener esta información a nuestra disposición durante el desarrollo del análisis, crearemos un archivo .txt con la salida anterior para poder recurrir a el cuando sea necesario.

In [285]:
# Escribir la lista de variables y descripciones en el archivo TXT
with open("data/descripciones_columnas", 'w', encoding='utf-8') as archivo_descripciones:
    # Escribir cada línea de variables y descripciones
    for variable, descripcion in descripciones:                 # Recorremos la lista creada anteriormente
        archivo_descripciones.write(f'{variable} : {descripcion}\n')

De esta forma hemos reducido el número de columnas de 299 a 69

>  **4.**

In [286]:
students.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 31651 entries, 0 to 31650
Data columns (total 70 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   IDENT           31651 non-null  int64 
 1   SEXO            31651 non-null  int64 
 2   EDAD            31651 non-null  int64 
 3   NACIO           31651 non-null  int64 
 4   TITU            31651 non-null  int64 
 5   RAMA            31651 non-null  int64 
 6   AMBITO          31651 non-null  int64 
 7   T_UNIV          31651 non-null  int64 
 8   DISCA           31651 non-null  int64 
 9   ESTUDIOS_PADRE  31651 non-null  int64 
 10  ESTUDIOS_MADRE  31651 non-null  int64 
 11  EST_B8          31651 non-null  int64 
 12  SAT1            31651 non-null  int64 
 13  SAT2            31651 non-null  object
 14  EST_B11_1       31651 non-null  int64 
 15  EST_B13_AMB1    31651 non-null  object
 16  EST_B11_2       31651 non-null  int64 
 17  EST_B11_3       31651 non-null  int64 
 18  EST_B1

Como podemos observar, ninguna columna tiene datos nulos de los 31651. Sin embargo, a simple vista podemos observar campos vacíos. Esto se puede deber a que los valores nulos están contemplados como string con espacios en blanco

In [287]:
students.iloc[0]['NOTRAB_BUS'] # Los registros sin datos, son espacios en blanco


' '

In [288]:
# Podríamos reemplazar estos valores por datos NaN. Sin embargo, más adelante lo realizaremos a la vez que modificamos el tipo de los datos, de un solo paso.
# students.replace(' ', pd.NA, inplace=True)

Otra observación a tener en cuenta es el tipo de datos. A primera vista, todos parecen ser numéricos; sin embargo, se observa que varias columnas están etiquetadas como 'object'. Esto podría atribuirse a lo mencionado anteriormente, donde los valores sin datos se representan como espacios en blanco. Optamos por una transformación que asegure que todos los valores sean tratados como numéricos. Tenemos certeza sobre el diseño de la base de datos y estamos seguros de que todos los datos son numéricos, sin riesgo de pérdida de información.

In [289]:
students.dtypes

IDENT           int64
SEXO            int64
EDAD            int64
NACIO           int64
TITU            int64
                ...  
PR_SUELDO      object
HATR_E27       object
NOTRAB_BUS     object
NOTRAB_BUSM    object
NOTRAB_BUST    object
Length: 70, dtype: object

In [290]:
students = students.apply(pd.to_numeric, errors='coerce', downcast='integer') 

# Al forzar el cambio a tipo numérico con errors='coerce', aquellos valores no numéricos (como los espacios en blanco), serán considerados NaN
# De esta forma realizamos dos pasos en 1: cambiar todos los valores a tipo numérico y modificar los string vacíos a nulos.

In [291]:
students.dtypes

IDENT            int16
SEXO              int8
EDAD              int8
NACIO             int8
TITU             int32
                ...   
PR_SUELDO      float64
HATR_E27       float64
NOTRAB_BUS     float64
NOTRAB_BUSM    float64
NOTRAB_BUST    float64
Length: 70, dtype: object

>  **5.**

El hecho de que todas las columnas sean numéricas se debe a que es un conjunto de datos codificado, es decir, cada número corresponde a un estado, una descripción que aparece en un archivo aparte que contiene toda la información de diseño de la tabla principal.

**PENDIENTE** Decodificación de variables. Crear diccionario. 

In [292]:
hoja_nombre = cod.loc[cod['Variable'] == 'SEXO', 'Diccionario ubicado en la hoja…'].values[0]
hoja_nombre

KeyError: 'Variable'

In [293]:
df_codificacion = pd.read_excel('data/dr_EILU_2019.xlsx', sheet_name=hoja_nombre,header=2)
df_codificacion
#info_variable = df_codificacion.loc[df_codificacion.iloc[:, 0] == variable, [0, 1]]

Unnamed: 0                                         Unnamed: 1  \
0          NaN                                                NaN   
1        TSEXO                                                NaN   
2       Código                                        Descripción   
3            1                                             Hombre   
4            2                                              Mujer   
..         ...                                                ...   
354          2  Sí, se lo ha planteado y se iría a vivir fuera...   
355          3  Sí, se lo ha planteado pero ha decidido quedar...   
356          4                          No, no se lo ha planteado   
357          9                                              NS/NC   
358                                                  No aplicable   

                        Unnamed: 2  
0    En hoja -Diseño-. Variables:   
1                             SEXO  
2                              NaN  
3                              NaN  
4                              NaN  
..                             ...  
354                            NaN  
355                            NaN  
356                            NaN  
357                            NaN  
358                            NaN  

[359 rows x 3 columns]

In [294]:
nombre_variable = df_codificacion.iloc[indice_variable, 0]

NameError: name 'indice_variable' is not defined

In [ ]:
    nombre_variable = df_codificacion.iloc[indice_variable, 0]
    
    # Encontrar la posición de la siguiente variable o el final del DataFrame
    siguiente_variable = df_codificacion.iloc[indice_variable + 1:].index[df_codificacion.iloc[indice_variable + 1:, 0].str.isnumeric()].min()

In [ ]:
# Función para procesar cada variable
def procesar_variable(df_codificacion, indice_variable):
    # Obtener el nombre de la variable
    nombre_variable = df_codificacion.iloc[indice_variable, 0]
    
    # Encontrar la posición de la siguiente variable o el final del DataFrame
    siguiente_variable = df_codificacion.iloc[indice_variable + 1:].index[df_codificacion.iloc[indice_variable + 1:, 0].str.isnumeric()].min()
    
    # Filtrar las filas entre el nombre de la variable y la siguiente variable
    df_variable = df_codificacion.iloc[indice_variable + 1:siguiente_variable, :]
    
    # Crear el diccionario con códigos y descripciones
    diccionario_variable = dict(zip(df_variable.iloc[:, 0], df_variable.iloc[:, 1]))
    
    return nombre_variable, diccionario_variable

# Leer la hoja de codificación
df_codificacion = pd.read_excel('tu_archivo.xlsx', sheet_name='NombreDeTuHojaDeCodificacion')

# Inicializar un diccionario para almacenar los resultados
diccionario_general = {}

# Iterar sobre las filas del DataFrame de codificación
for i, valor in enumerate(df_codificacion.iloc[:, 0]):
    # Procesar solo las filas que contienen nombres de variables
    if not pd.isna(valor) and not valor.isnumeric():
        nombre_variable, diccionario_variable = procesar_variable(df_codificacion, i)
        diccionario_general[nombre_variable] = diccionario_variable

# Mostrar el diccionario resultante
print(diccionario_general)

In [ ]:
# Iterar sobre las variables y construir el diccionario
diccionario_codificacion = {}

for variable in variables:
    # Paso 2: Obtener el nombre de la hoja correspondiente
    hoja_nombre = cod.loc[cod['Variable'] == variable, 'Diccionario ubicado en la hoja…'].values[0]
    
    # Paso 3: Acceder a la hoja y obtener la información de codificación
    df_codificacion = pd.read_excel('data/dr_EILU_2019.xlsx', sheet_name=hoja_nombre)
    info_variable = df_codificacion.loc[df_codificacion.iloc[:, 0] == variable, [0, 1]]
    
    # Paso 4: Crear el diccionario
    codificacion_dict = dict(zip(info_variable.iloc[:, 0], info_variable.iloc[:, 1]))
    
    diccionario_codificacion[variable] = codificacion_dict

# Mostrar el diccionario resultante
print(diccionario_codificacion)

TypeError: list indices must be integers or slices, not float

In [ ]:
cod

Variable Diccionario de la variable  Longitud Tipo  Decimales  Posición  \
0      IDENT                        NaN         5    N        NaN       1.0   
1       SEXO                      TSEXO         1    A        NaN       6.0   
2       EDAD                      TEDAD         1    A        NaN       7.0   
3      NACIO                     TNACIO         1    A        NaN       8.0   
4     NACIO1                    TNACIOP         1    A        NaN       9.0   
..       ...                        ...       ...  ...        ...       ...   
295  DURCUR6                    TDrCurs         1    A        NaN     371.0   
296  RESCUR6                    TReCurs         1    A        NaN     372.0   
297  ANHCUR6                    TAnCurs         4    A        NaN     373.0   
298   FACTOR                        NaN        11    N        6.0     377.0   
299    TOTAL                        NaN       387  NaN        NaN       NaN   

     Orden Diccionario ubicado en la hoja…  \
0      1.0                             NaN   
1      2.0                         Tablas1   
2      3.0                         Tablas1   
3      4.0                         Tablas1   
4      5.0                         Tablas1   
..     ...                             ...   
295  296.0                         Tablas2   
296  297.0                         Tablas2   
297  298.0                         Tablas2   
298  299.0                             NaN   
299    NaN                             NaN   

                                           Descripción  
0                   Nº de identificación de la persona  
1                                                 Sexo  
2                        Grupo de edad (a 31-dic-2019)  
3                                         Nacionalidad  
4                              País de nacionalidad 1   
..                                                 ...  
295  Duración del sexto curso de formación del SEPE...  
296  Resultado del sexto curso de formación del SEP...  
297  Año de finalización del sexto curso de formaci...  
298              Factor de elevación (seis decimales )  
299                                                NaN  

[300 rows x 9 columns]

> **6**

Una vez hemos realizado la transformación de los datos, modificaremos pequeñas características con el fin de trabajar de una forma más cómoda con los datos. Cambiaremos el nombre de 

> **7** Exploratory Data Anaysis

Para comenzar el análisis de las variables, es necesario comprender exactamente qué estamos buscando, con el fin de poder seleccionar aquellos datos que nos aporten mayor información para nuestra hipótesis. 

Nuestro objetivo principal es obtener aquellos factores determinantes que influyen a la hora de encontrar trabajo para un estudiante universitario:

- Estudiar en una universidad privada te garantiza más oportunidades de empleo al terminar el grado universitario

- ¿La educación familiar influye en la empleabilidad del estudiante?

- ¿Los idiomas son importantes a la hora de buscar trabajo?

- ¿Los estudiantes de ciertas ramas tiene más oportunidades de encontrar trabajo?

### UNIVERSIDAD PÚBLICA VS PRIVADA


Para una primera observación, filtraremos a aquellos individuos que han estudiado únicamente un grado universitario, con el fin de obtener una respuesta más acertada a nuestra hipótesis inicial, ya que tener más de un título impediría diferenciar cuál de ellos ha sido determinante para obtener el empleo.  

Las columnas que nos aportan esta información son las siguientes*:



*Primer grupo*
- **EST_B11_1** : Otros estudios: Grado / Diplomatura / Licenciatura o equivalente
- **EST_B11_2** : Otros estudios: Máster universitario
- **EST_B11_3** : Otros estudios: Doctorado universitario
- **EST_B11_4** : Otros estudios: Estudios artísticos superiores (artes plásticas, diseño, música y danza, arte dramático o conservación y restauración de bienes culturales)
- **EST_B11_5**: Otros estudios: Ciclos Formativos de grado superior (formación profesional, artes plásticas y diseño) o Enseñanzas Deportivas de Grado Superior
- **EST_B11_6**: Otros estudios: Ciclos Formativos de grado medio (formación profesional, artes plásticas y diseño) o Enseñanzas Deportivas de Grado Medio

*Segundo grupo*
- **EST_B12** : Nº de otros estudios: Grado / Diplomatura / Licenciatura o equivalente
- **EST_B16** : Nº de otros estudios: Máster universitario
- **EST_B24** : Nº de otros estudios: Ciclos Formativos de grado superior (formación profesional, artes plásticas y diseño) o Enseñanzas Deportivas de Grado Superior
- **EST_B28** : Nº de otros estudios: Ciclos Formativos de grado medio (formación profesional, artes plásticas y diseño) o Enseñanzas Deportivas de Grado Medio

**Obtenido a partir del archivo 'data/descripciones_columnas.txt' creado anteriormente*

En el **primer grupo** se indica con variables categóricas (1 -> SÍ ; 2 -> No) si el individuo ha cursado alguno de los siguientes 6 títulos: grado, máster, doctorado, estudios artísticos, grado superior o grado medio.

Las variables del **segundo grupo** indican el número de títulos de grado, máster, grado superior o grado medio que tiene el individuo.

Hay que tener en cuenta que en el segundo grupo no se contabilizan los estudios artísticos superiores ni el doctorado universitario. Por esta razón, filtraremos por todas las variables del primer grupo para asegurarnos que la muestra con la que vamos a realizar el análisis solo tiene un grado universitario.

In [305]:
# Se excluyen todos los registros que contenga otros estudios de cualquier tipo 
# Son variables categóricas donde el valor 2 indica NO

students[(students['EST_B11_1']==2) & (students['EST_B11_2']==2) & (students['EST_B11_3']==2) &
         (students['EST_B11_4']==2) & (students['EST_B11_5']==2) & (students['EST_B11_6']==2)]


IDENT  SEXO  EDAD  NACIO    TITU  RAMA  AMBITO  T_UNIV  DISCA  \
2          3     1     1      1   11101     3      11       1      2   
3          4     1     1      1   11101     3      11       1      2   
7          8     2     1      1   11101     3      11       1      2   
14        15     2     1      1   11101     3      11       1      2   
15        16     2     1      1   11101     3      11       1      2   
...      ...   ...   ...    ...     ...   ...     ...     ...    ...   
31631  31632     1     2      1  109999     3     109       1      2   
31632  31633     1     1      1  109999     3     109       1      2   
31640  31641     1     1      1  109999     3     109       1      2   
31641  31642     2     1      1  109999     3     109       1      2   
31643  31644     1     1      1  109999     3     109       1      2   

       ESTUDIOS_PADRE  ...  PR_SIT_PR  PR_JOR  PR_NIVEL  PR_AREA  PR_CONOC  \
2                   3  ...        3.0     2.0       4.0      3.0       2.0   
3                   7  ...        3.0     2.0       4.0      3.0       2.0   
7                   4  ...        1.0     2.0       7.0      3.0       2.0   
14                  4  ...        3.0     2.0       7.0      4.0       2.0   
15                  2  ...        3.0     2.0       7.0      4.0       2.0   
...               ...  ...        ...     ...       ...      ...       ...   
31631               3  ...        3.0     1.0       7.0      4.0       2.0   
31632               8  ...        3.0     1.0       4.0      2.0       1.0   
31640               5  ...        2.0     2.0       4.0      3.0       2.0   
31641               5  ...        5.0     1.0       3.0      3.0       2.0   
31643               3  ...        3.0     1.0       6.0      3.0       2.0   

       PR_SUELDO  HATR_E27  NOTRAB_BUS  NOTRAB_BUSM  NOTRAB_BUST  
2            3.0       1.0         NaN          NaN          NaN  
3            2.0       1.0         NaN          NaN          NaN  
7            2.0       2.0         NaN          NaN          NaN  
14           3.0       2.0         NaN          NaN          NaN  
15           3.0       1.0         NaN          NaN          NaN  
...          ...       ...         ...          ...          ...  
31631        1.0       2.0         NaN          NaN          NaN  
31632        2.0       1.0         NaN          NaN          NaN  
31640        3.0       2.0         NaN          NaN          NaN  
31641        1.0       2.0         NaN          NaN          NaN  
31643        1.0       1.0         NaN          NaN          NaN  

[9326 rows x 70 columns]

Ahora el conjunto de datos se ha reducido considerablemente, a 9326 registros.